improver/regrid/landsea.py

```diff
@@ -271,46 +272,67 @@ def correct_where_input_true(self, selector_val: int) -> None:
                 Intended to be 1 for filling land points near the coast
                 and 0 for filling sea points near the coast.
         """
-        # Find all points on output grid matching selector_val
-        use_points = np.where(self.input_land.data == selector_val)
-
-        # If there are no matching points on the input grid, no alteration can
-        # be made. This tests the size of the y-coordinate of use_points.
-        if use_points[0].size == 0:
-            return
-
-        # Get shape of output grid
-        ynum, xnum = self.output_land.shape
-
-        # Using only these points, extrapolate to fill domain using nearest
-        # neighbour. This will generate a grid where the non-selector_val
-        # points are filled with the nearest value in the same mask
-        # classification.
-        (y_points, x_points) = np.mgrid[0:ynum, 0:xnum]
-        selector_data = griddata(
-            use_points,
-            self.nearest_cube.data[use_points],
-            (y_points, x_points),
-            method="nearest",
-        )
-
-        # Identify nearby points on regridded input_land that match the
-        # selector_value
-        if selector_val > 0.5:
-            thresholder = BasicThreshold(0.5)
-        else:
-            thresholder = BasicThreshold(0.5, comparison_operator="<=")
-        in_vicinity = self.vicinity(thresholder(self.input_land))
-
-        # Identify those points sourced from the opposite mask that are
-        # close to a source point of the correct mask
-        mismatch_points = np.logical_and(
-            np.logical_and(
-                self.output_land.data == selector_val,
-                self.input_land.data != selector_val,
-            ),
-            in_vicinity.data > 0.5,
-        )
+        if selector_val not in self._cache_nn:
+            # Find all points on output grid matching selector_val
+            use_points = np.where(self.input_land.data == selector_val)
+
+            # If there are no matching points on the input grid, no alteration can
+            # be made. This tests the size of the y-coordinate of use_points.
+            if use_points[0].size == 0:
+                return None, None
+
+            # Get shape of output grid
+            ynum, xnum = self.output_land.shape
+
+            # # Using only these points, extrapolate to fill domain using nearest
+            # # neighbour. This will generate a grid where the non-selector_val
+            # # points are filled with the nearest value in the same mask
+            # # classification.
+            (y_points, x_points) = np.mgrid[0:ynum, 0:xnum]
+
+            no_use_points = np.where(self.input_land.data != selector_val)
```

Comment:
I think the `no_use_points` should go below the `use_points`, is it faster to `np.where` again here or invert the selection?

Reply:
Can't revert indices (`np.where`) would have to switch to using masks.  I did this but then seeing your comment above concerning `y_points[use_points]` -> `use_points[0]` I have opted to go back to using `np.where` after all